graylog2-web-interface/src/views/components/searchbar/queryvalidation/QueryValidation.tsx

```diff
@@ -182,6 +183,36 @@ const deduplicateExplanations = (explanations: Explanations | undefined): Explan
   return deduplicated;
 };
 
+const warmTierExplanations = (searchesWarmTier: boolean, warmTierIndices: Array<IndexRange>) => {
```

Comment:
I think we should move the warm tier logic to its own component and use `Plugin` system. This component already supports it, with the `views.elements.validationErrorExplanation` key.